### ISSUE TITLE
Yara 4.1.3 make check fails on test-rules and test-exception

### ISSUE THREAD

**Issue by aps-kar** [VirusTotal/yara#1619]
Following instructions from the docs, initially I ran:

./bootstrap.sh
./configure
make
sudo make install

And then:

make check

Which resulted in the following errors:

![image](https://user-images.githubusercontent.com/30056014/146364710-11124455-a062-43da-8c6b-cf1e8223ebac.png)

I'm running Kali over WSL2, latest kernel version: 5.10.60.1-microsoft-standard-WSL2 x86_64.

Please let me know if there is any other information I can provide.

**Comment by plusvic** [comment]
It looks like the `reexec` function (https://github.com/VirusTotal/yara/blob/master/tests/test-exception.c#L222) in the test is not working as expected in WSL2. It could be either `fork` returning a negative number, `execve` returning non-zero or `waitpid` setting a non-zero status. It would be helpful if you can add some `printf` statements in `reexec` for determining at which point `reexec` is failing.

**Comment by plusvic** [comment]
So, `waitpid` is setting `status` to 256, which apparently means that the child process received a `SIGHUP` signal. @hillu, do you have any clue about this issue? 

The code is pretty old and nobody have complained before, so it looks like something that only happens in WSL, probably something that works different from other POSIX systems.

**Comment by hillu** [comment]
> So, `waitpid` is setting `status` to 256, which apparently means that the child process received a `SIGHUP`

No, the child ended with `exit(1)`. I don't have an idea why, though.

I'm not really sure what is going on here and I don't have a Windows box with WSL2 available for testing. Does the Linux environment share filesystems or does it run off its own block device?

**Comment by aps-kar** [comment]
> Does the Linux environment share filesystems or does it run off its own block device?

Not sure to be honest but I would lean on the former.
